### PR TITLE
coverity fixes

### PR DESCRIPTION
**Pull request by rgetz**
Fixes a few new coverity issues 

https://scan.coverity.com/projects/analogdevicesinc-libiio

More to come.

### CODE REVIEW

**Comment by commodo** [comment]
ummm; so it looks like coverity did not fire for these during review/build
is there a build issue with it that we should fix first?

**Comment by rgetz** [comment]
the free version of coverity only allows one or two reviews per day, so we can't do them on pull requests. We only do them on master.

File: context.c [hunk 1]
```diff
@@ -311,19 +312,18 @@ struct iio_context * iio_create_default_context(void)
...
 	if (hostname) {
-		struct iio_context *ctx;
+		if (strlen(hostname) > 2 && strlen(hostname) < MAXHOSTNAMELEN + 4) {
+			struct iio_context *ctx;
 
-		ctx = iio_create_context_from_uri(hostname);
-		if (ctx)
-			return ctx;
-
-#ifdef WITH_NETWORK_BACKEND
```
Comment (commodo) [inline context.c:320]: i'm having trouble following this;
is it ok to remove `#ifdef WITH_NETWORK_BACKEND` ?
is `#ifdef HAVE_DNS_SD` dependent on `#ifdef WITH_NETWORK_BACKEND`?

**Comment by commodo** [comment]
> the free version of coverity only allows one or two reviews per day, so we can't do them on pull requests. We only do them on master.

ack

**Comment by commodo** [comment]
only 1 comment;
rest looks good

File: context.c (hunk 1 above)
Comment (rgetz) [reply context.c:320]: yes - `HAVE_DNS_SD` can only be set if `WITH_NETWORK_BACKEND` is set, and it is more appropriate for this code path.

Comment (rgetz) [reply context.c:320]: `HAVE_DNS_SD` is what allows empty strings to return as IP numbers - without that - calls to the networking will just fail with empty strings.

Comment (rgetz) [reply context.c:320]: [CMakeLists.txt](https://github.com/analogdevicesinc/libiio/blob/master/CMakeLists.txt#L235-L286)

`HAVE_DNS_SD` is set inside the `if(WITH_NETWORK_BACKEND)`

**Comment by rgetz** [comment]
Just as FYI, this did clean up 7 issues.

image

**Pull request merged**